test/test_cartesian_product.cpp

```diff
@@ -17,9 +24,59 @@ namespace {
 
 constexpr bool test_cartesian_product()
 {
+    // 1D `cartesian_product`.
+    {
+        auto cart = flux::cartesian_product(std::array{100, 200, 300});
+
+        using C = decltype(cart);
+
+        static_assert(flux::sequence<C>);
+        static_assert(flux::multipass_sequence<C>);
+        static_assert(flux::bidirectional_sequence<C>);
+        static_assert(flux::random_access_sequence<C>);
+        static_assert(not flux::contiguous_sequence<C>);
+        static_assert(flux::bounded_sequence<C>);
+        static_assert(flux::sized_sequence<C>);
+
+        static_assert(flux::sequence<C const>);
+        static_assert(flux::multipass_sequence<C const>);
+        static_assert(flux::bidirectional_sequence<C const>);
+        static_assert(flux::random_access_sequence<C const>);
+        static_assert(not flux::contiguous_sequence<C const>);
+        static_assert(flux::bounded_sequence<C const>);
+        static_assert(flux::sized_sequence<C const>);
+
+        static_assert(std::same_as<flux::element_t<C>, std::tuple<int&>>);
+        static_assert(std::same_as<flux::value_t<C>, std::tuple<int>>);
+        static_assert(std::same_as<flux::rvalue_element_t<C>, std::tuple<int&&>>);
+
+        // FIXME: These currently fail - they're actually `tuple<const int>` etc. Is that correct?
+        //static_assert(std::same_as<flux::element_t<C const>, std::tuple<int&>>);
+        //static_assert(std::same_as<flux::value_t<C const>, std::tuple<int>>);
+        //static_assert(std::same_as<flux::rvalue_element_t<C const>, std::tuple<int&&>>);
```

Comment:
@tcbrindle I added a test for a 1D `cartesian_product` and was surprised that these tests failed. Thoughts?